app/controllers/samples_controller.rb

```diff
@@ -533,7 +539,16 @@ def samples_going_public
       params[:userId] ? User.find(params[:userId]) : current_user,
       params[:projectId] ? Project.find(params[:projectId]) : nil
     )
-    render json: samples.to_json(include: [{ project: { only: [:id, :name] } }])
+
+    # Efficiently fetch auxiliary information from Project model
+    samples_resp = []
+    samples.includes(:project).each do |s|
```

Comment:
Should this be moved to a helper method to retrieve `project properties`? 
Since it is used above too.